src/components/YearCalendar.vue

```diff
@@ -106,14 +130,34 @@ export default {
         .format('YYYY-MM-DD')
       this.$emit('toggleDate', {
         date: activeDate,
-        selected: dateObj.selected
+        selected: dateObj.selected,
+        className: dateObj.className
       })
+
       let newDates = [...this.activeDates]
-      let dateIndex = newDates.indexOf(activeDate)
-      if (dateIndex === -1) {
-        newDates.push(activeDate)
+
+      if (this.activeDates.length && typeof this.activeDates[0] === 'string') {
```

Comment:
Hi Nono. Yes, indeed, that happens because after the the array is empty, there is no way to know the initial state (if it was an array of strings, or an array of objects). A new prop needs to be created, but I believe that it is what you propose on your next comment.